Kernel/Memory/MemoryManager.cpp

```diff
@@ -518,31 +520,27 @@ UNMAP_AFTER_INIT void MemoryManager::parse_memory_map_fdt(MemoryManager::GlobalD
                     if (property_name == "reg"sv) {
                         VERIFY(state.address_cells);
                         VERIFY(state.size_cells);
-                        // FIXME: We may get more than one range here
-                        if (data.size() > (state.address_cells + state.size_cells) * sizeof(u32))
-                            TODO();
-                        if (state.address_cells == 1) {
-                            BigEndian<u32> data_as_int;
-                            __builtin_memcpy(&data_as_int, data.data(), sizeof(u32));
-                            state.start = data_as_int;
-                            data = data.slice(sizeof(u32));
-                        } else {
-                            BigEndian<u64> data_as_int;
-                            __builtin_memcpy(&data_as_int, data.data(), sizeof(u64));
-                            state.start = data_as_int;
-                            data = data.slice(sizeof(u64));
-                        }
 
-                        if (state.size_cells == 1) {
-                            BigEndian<u32> data_as_int;
-                            __builtin_memcpy(&data_as_int, data.data(), sizeof(u32));
-                            state.size = data_as_int;
-                            data = data.slice(sizeof(u32));
-                        } else {
-                            BigEndian<u64> data_as_int;
-                            __builtin_memcpy(&data_as_int, data.data(), sizeof(u64));
-                            state.size = data_as_int;
-                            data = data.slice(sizeof(u64));
+                        FixedMemoryStream reg_stream { data };
+
+                        while (!reg_stream.is_eof()) {
```

Comment:
Can we pre-allocate the state.reg here, as the eventual size is quite easy to calculate